snuba/settings/__init__.py

```diff
@@ -278,6 +255,46 @@
 COUNTER_WINDOW_SIZE = timedelta(minutes=10)
 
 
+# Slicing Configuration
+
+# Mapping of storage key to slice count
+# This is only for sliced storages
+SLICED_STORAGES: Mapping[str, int] = {}
+
+# Mapping storage key to a mapping of logical partition
+# to slice id
+LOGICAL_PARTITION_MAPPING: Mapping[str, Mapping[int, int]] = {}
+
+# The slice configs below are the "SLICED" versions to
+# the equivalent default settings above. For example,
+# "SLICED_KAFKA_TOPIC_MAP" is the "SLICED" version of
+# "KAFKA_TOPIC_MAP". These should be filled out
+# for any corresponding sliced storages defined above,
+# with the applicable number of slices in mind.
+
+
+# Storage set keys should be defined either in CLUSTERS
+# or SLICED_CLUSTERS. CLUSTERS will define clusters
+# which are not sliced, i.e. are associated with
+# only the default slice_id (0). CLUSTERS is defined in
+# the default way, without adding slice id in
+# the storage_sets field. We define sliced clusters,
+# i.e. clusters that reside on multiple slices
+# in SLICED_CLUSTERS. We define all associated
+# (storage set, slice id) pairs in SLICED_CLUSTERS
+# in the storage_sets field. Other fields are defined
+# in the same way as they are in CLUSTERS.
+SLICED_CLUSTERS: Sequence[Mapping[str, Any]] = []
+
+# Mapping of (logical topic names, slice id) pairs to custom physical topic names
+# This is only for sliced Kafka topics
+SLICED_KAFKA_TOPIC_MAP: Mapping[Tuple[str, int], str] = {}
+
+# Mapping of physical Kafka topic names to broker config
+# This is only for sliced Kafka topics
+SLICED_KAFKA_BROKER_CONFIG: Mapping[str, Mapping[str, Any]] = {}
```

Comment:
Don't these all need the sliced topic in the key like `SLICED_KAFKA_TOPIC_MAP` has? e.g. `SLICED_KAFKA_BROKER_CONFIG: Mapping[Tuple[str, int], Mapping[str, Any]] = {}`. Otherwise how do you know which slice the config is for?

Reply:
So the way it currently works is `SLICED_KAFKA_TOPIC_MAP` maps something like ("events", 1) to "events-1". And `SLICED_KAFKA_BROKER_CONFIG` maps "events-1" to [broker config].

Reply:
I don't think this is the right approach. Reasons:
1. It is very fragile. It assumes that the user has assigned unique topic names on all slices and breaks if they reused the same default name (which imo is the normal thing to do). Tuple of default topic and slice ID is guaranteed to be unique so is a much better choice to me
2. It basically now works the opposite of KAFKA_BROKER_CONFIG where the map is based on the default topic name. This is unintuitive as one would expect `SLICED_KAFKA_BROKER_CONFIG` to operate more like `KAFKA_BROKER_CONFIG`

Reply:
If I'm understanding correctly: no `SLICED_KAFKA_TOPIC_MAP`, and changing the keys of `SLICED_KAFKA_BROKER_CONFIG`. So for example:

`SLICED_KAFKA_BROKER_CONFIG` = {("snuba-generic-metrics", 1), BROKER_CONFIG}

How does this look to you?

Reply:
Yes, this is what I was thinking. This is exactly how KAFKA_TOPIC_MAP works. The only extra bit is the 1.

Reply:
No, that's not exactly what I was suggesting. You still probably want to keep the ability to remap all the topic names per slice so you want to keep the `SLICED_KAFKA_TOPIC_MAP` for that. But yes, change the `SLICED_KAFKA_BROKER_CONFIG` to that.

Reply:
Ok, understood. Making that change now!